enumerables.rb

```diff
@@ -0,0 +1,139 @@
+module Enumerable
+  # 1. Create my_each
+  def my_each
+    (0..length - 1).each do |i|
+      yield self[i]
+    end
+  end
+
+  # 2. Create my_each_with_index
+  def my_each_with_index
+    (0..length - 1).each do |i|
+      yield(self[i], i)
+    end
+  end
+
+  # 3. Create my_select
+  def my_select
+    arr = []
+    (0..length - 1).each do |i|
+      criteria = yield(self[i])
+      arr << self[i] if criteria
+    end
+    arr
+  end
+
+  # 4. Create my_all?
+  def my_all?
+    arr = []
+    my_each do |i|
+      condition = yield(i)
+      arr << (condition ? true : false)
+    end
+    arr.include?(false) ? false : true
+  end
+
+  # 5. Create my_any?
+  def my_any?
+    arr = []
+    my_each do |i|
+      condition = yield(i)
+      arr << (condition ? true : false)
+    end
+    arr.include?(true)
+  end
+
+  # 6. Create my_none?
+  def my_none?
+    arr = []
+    my_each do |i|
+      condition = yield(i)
+      arr << (condition ? true : false)
+    end
+    arr.include?(true) ? false : true
+  end
+
+  # 7. Create my_count
+  def my_count
+    counter = 0
+    each do |i|
+      condition = yield(i)
+      counter += 1 if condition
+    end
+    counter
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] returns the number of items in enum through enumeration
```ruby
array.my_count == array.count #true
```
- [ ] counts the number of items in enum that are equal to item if an argument is given
```ruby
array.my_count(0) == array.count(0) #true
```